exporters/otlp/lib/opentelemetry/exporters/otlp/exporter.rb

```diff
@@ -0,0 +1,252 @@
+        #   list of recorded {OpenTelemetry::SDK::Trace::SpanData} structs to be
+        #   exported.
+        # @return [Integer] the result of the export.
+        def export(span_data)
+          return FAILURE if @shutdown
+
+          send_bytes(encode(span_data))
+        end
+
+        # Called when {OpenTelemetry::SDK::Trace::Tracer#shutdown} is called, if
+        # this exporter is registered to a {OpenTelemetry::SDK::Trace::Tracer}
+        # object.
+        def shutdown
+          @shutdown = true
+          @http.finish if @http.started?
+        end
+
+        private
+
+        def send_bytes(bytes)
+          retry_count = 0
+          untraced do
+            request = Net::HTTP::Post.new(@path)
+            request.body = bytes
+            request.add_field('Content-Type', 'application/x-protobuf')
+            # TODO: enable gzip when https://github.com/open-telemetry/opentelemetry-collector/issues/1344 is fixed.
+            # request.add_field('Content-Encoding', 'gzip')
+  
+            @http.start unless @http.started?
+            response = @http.request(request)
+
+            case response
+            when Net::HTTPOK
+              response.body # Read and discard body
+              SUCCESS
+            when Net::HTTPServiceUnavailable, Net::HTTPTooManyRequests
+              response.body # Read and discard body
+              redo if backoff?(retry_after: response['Retry-After'], retry_count: retry_count += 1)
+              FAILURE
+            when Net::HTTPRequestTimeOut, Net::HTTPGatewayTimeOut, Net::HTTPBadGateway
+              response.body # Read and discard body
+              redo if backoff?(retry_count: retry_count += 1)
+              FAILURE
+            when Net::HTTPBadRequest, Net::HTTPClientError, Net::HTTPServerError
+              # TODO: decode the body as a google.rpc.Status Protobuf-encoded message when https://github.com/open-telemetry/opentelemetry-collector/issues/1357 is fixed.
+              response.body # Read and discard body
+              FAILURE
+            when Net::HTTPRedirection
+              @http.finish
+              handle_redirect(response['location'])
+              redo if backoff?(retry_after: 0, retry_count: retry_count += 1)
+            else
+              @http.finish
+              FAILURE
+            end
+          rescue Net::OpenTimeout, Net::ReadTimeout
+            retry if backoff?(retry_count: retry_count += 1)
+            return FAILURE
+          end
+        end
+
+        def handle_redirect(location)
+          # TODO: figure out destination and reinitialize @http and @path
+          location = response['location']
+          location = URI(location).relative? ? (@uri + location).to_s : location
+          initialize_http(location) # Assume this is our new destination from now on.
+        end
+
+        def untraced
+          @tracer.with_span(OpenTelemetry::Trace::Span.new) { yield }
```

Comment:
This implements "untracing". It is only effective if the active sampler respects the parent's sampling decision. We create a parent span that is not sampled and not recorded (and, incidentally, not finished). Child spans within this block should also be not sampled and not recorded.

Reply:
We discussed this briefly at the last SIG meeting and I just made an issue, but we should be able to implement an untraced method using context: https://github.com/open-telemetry/opentelemetry-ruby/issues/313. This doesn't necessarily need to happen in this PR though.

Reply:
👍 I’ll stick with the current method for this PR. We can work out the context-based mechanism in #313.